test/integration/puppetlabs/services/certificate_authority/certificate_authority_int_test.clj

```diff
@@ -968,90 +1021,164 @@
+           :certificate-authority
+           {:allow-auto-renewal true}}
+          (let [response (http-client/post
+                           "https://localhost:8140/puppet-ca/v1/certificate_renewal"
+                           {:ssl-cert    (str bootstrap/server-conf-dir "/ssl/certs/localhost.pem")
+                            :ssl-key     (str bootstrap/server-conf-dir "/ssl/private_keys/localhost.pem")
+                            :ssl-ca-cert (str bootstrap/server-conf-dir "/ca/ca_crt.pem")
+                            :as          :text})]
+            (is (= 501 (:status response))))))
+
+      (testing "returns a 400 bad request response when the ssl-client-cert is not present"
+        (bootstrap/with-puppetserver-running-with-mock-jrubies
+          "JRuby mocking is safe here because all of the requests are to the CA
+          endpoints, which are implemented in Clojure."
+          app
+          {:jruby-puppet
+           {:gem-path [(ks/absolute-path jruby-testutils/gem-path)]}
+           :webserver
+           {:ssl-cert     (str bootstrap/server-conf-dir "/ssl/certs/localhost.pem")
+            :ssl-key      (str bootstrap/server-conf-dir "/ssl/private_keys/localhost.pem")
+            :ssl-ca-cert  (str bootstrap/server-conf-dir "/ca/ca_crt.pem")
+            :ssl-crl-path (str bootstrap/server-conf-dir "/ssl/crl.pem")}
+           :certificate-authority
+           {:allow-auto-renewal true}}
+          (let [response (http-client/post
+                           "https://localhost:8140/puppet-ca/v1/certificate_renewal"
+                           {:ssl-ca-cert (str bootstrap/server-conf-dir "/ca/ca_crt.pem")
+                            :as          :text})]
+            (is (= 400 (:status response)))
+            (is (= "No certificate found in renewal request" (:body response)))))))
+
+    (testing "with allow-header-cert-info = true"
+      (testing "returns a 501 not implemented response when the feature is enabled,
+            a cert is supplied in header and the signing certificate matches"
+        (bootstrap/with-puppetserver-running-with-mock-jrubies
+          "JRuby mocking is safe here because all of the requests are to the CA
+          endpoints, which are implemented in Clojure."
+          app
+          {:jruby-puppet
+           {:gem-path [(ks/absolute-path jruby-testutils/gem-path)]}
+           :webserver
+           {:ssl-cert     (str bootstrap/server-conf-dir "/ssl/certs/localhost.pem")
+            :ssl-key      (str bootstrap/server-conf-dir "/ssl/private_keys/localhost.pem")
+            :ssl-ca-cert  (str bootstrap/server-conf-dir "/ca/ca_crt.pem")
+            :ssl-crl-path (str bootstrap/server-conf-dir "/ssl/crl.pem")}
+           :certificate-authority
+           {:allow-auto-renewal true}
+           :authorization
+           {:allow-header-cert-info true}}
+          (let [header-cert (ring-codec/url-encode (slurp (str bootstrap/server-conf-dir "/ca/ca_crt.pem")))
```

Comment:
this should be localhost.pem (or whatever we're using as the "agent' cert) right?

Reply:
good catch, fixed